bin/analysis.cpp

```diff
@@ -29,17 +29,40 @@
 #include <unordered_set>
 
 #define _ENDSTRING std::string::npos
-#define DEBUG
+//#define DEBUG
+//#define DEBUGG
 
 /**
 TT
-DY TANDP POWHEG AMC MAD
+DY TANDP POWHEG AMC MAD POWINCL
 W
 WZ
 ZZ
 data TANDP
 */
 
+/// return the dataOverMC dilepton mass correction factor for one DY sample
+double getDyMllScaleFactor(std::string chnl, std::string mode_str, std::string inputFile){
+	std::string ch = "", dataset = "";
+	double scalefactor = 1.0;
+	std::ifstream readScaleFactors;
+	readScaleFactors.open(inputFile);
+	if(readScaleFactors.is_open() == false){
+		std::cerr << "[ERROR] File " << inputFile << " not opened. Check if the file exists" << std::endl;
+		return 1;
+	}
+
+	while(readScaleFactors.peek() != EOF && readScaleFactors.good()) {
+		readScaleFactors >> ch >> dataset >> scalefactor;
+#ifdef DEBUGG
+		std::cout<<"ch=\t"<< ch <<"\tdataset=\t"<< dataset <<"\tscalefactor=\t"<< scalefactor << std::endl;
+#endif
+		if(ch.compare(chnl) == 0 && mode_str.find(dataset) != _ENDSTRING) return scalefactor;
```

Comment:
If I understand correctly for each channel you read the file. So if you want to run analysis.cpp on multiple modes, this would imply reading the file multiple times....